src/main/java/uk/gov/digital/ho/proving/income/api/NinoUtils.java

```diff
@@ -1,20 +1,48 @@
 package uk.gov.digital.ho.proving.income.api;
 
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Component;
+
 import java.util.regex.Pattern;
 
+import static java.util.Objects.isNull;
+import static java.util.Objects.nonNull;
+
+@Component
 public class NinoUtils {
-    private NinoUtils() {
-        // utility
-    }
-    public static String sanitiseNino(String nino) {
-        return nino.replaceAll("\\s", "").toUpperCase();
+    private static final Pattern NINO_PATTERN = Pattern.compile("(^((?!(BG|GB|KN|NK|NT|TN|ZZ)|([DFIQUV])[A-Z]|[A-Z]([DFIOQUV]))[A-Z]{2})[0-9]{6}[A-D]?$)");
+
+    public String sanitise(final String nino) {
+        if (isNull(nino)) {
+            return null;
+        }
+        return StringUtils.deleteWhitespace(nino).toUpperCase();
     }
 
-    public static void validateNino(String nino) {
-        final Pattern pattern = Pattern.compile("^[a-zA-Z]{2}[0-9]{6}[a-dA-D]{1}$");
-        if (!pattern.matcher(nino).matches()) {
+    public void validate(final String nino) {
+        if (isInvalid(nino)) {
             throw new IllegalArgumentException("Error: Invalid NINO");
         }
     }
 
+    public String redact(final String nino) {
+        final String sanitisedNino = sanitise(nino);
+
+        if (isInvalid(sanitisedNino)) {
+            return nino;
+        }
+
+        final String firstFourCharacters = StringUtils.left(sanitisedNino, 4);
+        final String lastTwoCharacters = StringUtils.right(sanitisedNino, 2);
+
+        return firstFourCharacters + "***" + lastTwoCharacters;
```

Comment:
I know that the length of a nino will never change, but I don't like the hardcoded value "***"...  I'd rather we calculated its length... but then we'd have some complicated code to cater for something that will never change.  Gah please ignore this comment.

Reply:
New method of redaction - from Stuart & based on HMRC - does dynamically generate the length of  redaction now.